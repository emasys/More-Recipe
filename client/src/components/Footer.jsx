import React from 'react';

const Footer = () => (
  <div>
    <footer className="container-fluid bg-dark footer-main">
      <div className="container ">
        <div className="row">
          <div className="col-12 footer-social text-center mt-100">
            <p className="text-center text-white">
                &copy;2017 All rights reserved
            </p>
            <a href="https://web.facebook.com/emasys23">
              <i className="fa fa-facebook-square fa-2x" aria-hidden="true" />
            </a>&nbsp;
            <a href="https://twitter.com/emasys_nd">
              <i className="fa fa-twitter-square fa-2x" aria-hidden="true" />
            </a>&nbsp;
            <a href="#">
              <i className="fa fa-instagram fa-2x" aria-hidden="true" />
            </a>
          </div>
        </div>
      </div>
    </footer>
  </div>
);

export default Footer;
