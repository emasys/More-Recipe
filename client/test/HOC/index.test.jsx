import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';

import withHoc from '../../src/components/HOC/index';
import fakeStore, { unauthenticatedFakeStore } from '../__mocks__/fakeStore';
import Footer from '../../src/components/Footer';

let store = createMockStore(fakeStore);

describe('higher-order component', () => {
  let WrapperComponent;

  beforeEach(() => {
    WrapperComponent = withHoc(Footer);
  });

  test('withHoc should render without crashing', () => {
    const wrapper = mount(<Provider store={store}>
      <WrapperComponent />
    </Provider>);
    expect(wrapper).toMatchSnapshot();
  });

  test('withHoc should render with unauthenticated data without crashing', () => {
    store = createMockStore(unauthenticatedFakeStore);
    const props = {
      location: { pathname: '/signin' },
      history: { push: jest.fn() }
    };
    const wrapper = mount(<Provider store={store} >
      <WrapperComponent {...props} />
    </Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
