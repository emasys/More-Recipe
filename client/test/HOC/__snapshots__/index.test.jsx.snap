// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`higher-order component withHoc should render with unauthenticated data without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Authenticate)
    history={
      Object {
        "push": [Function],
      }
    }
    location={
      Object {
        "pathname": "/signin",
      }
    }
  >
    <Authenticate
      auth={
        Object {
          "allUsers": Object {
            "success": true,
            "users": Array [
              Object {
                "avatar": null,
                "bio": null,
                "country": null,
                "createdAt": "2018-03-13T05:04:37.443Z",
                "email": "emasysnd@gmail.com",
                "firstName": null,
                "id": 1,
                "lastName": null,
                "moniker": "admin",
                "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
                "updatedAt": "2018-03-13T05:04:37.443Z",
              },
            ],
          },
          "authInfo": Object {
            "isLoggedIn": true,
            "userId": 1,
            "username": "admin",
          },
          "isLoggedIn": false,
          "message": null,
          "userInfo": Object {
            "data": Object {
              "avatar": null,
              "bio": null,
              "country": "Nigeria",
              "email": "emasysnd@gmail.com",
              "firstName": "emmy",
              "id": 1,
              "lastName": "endy",
              "moniker": "admin",
            },
            "success": true,
          },
          "userProfile": Object {
            "data": Object {
              "avatar": null,
              "bio": null,
              "country": null,
              "email": "emasysnd@gmail.com",
              "firstName": null,
              "id": 1,
              "lastName": null,
              "moniker": "admin",
            },
            "success": true,
          },
        }
      }
      flashMessage={[Function]}
      history={
        Object {
          "push": [Function],
        }
      }
      isAuthenticated={[Function]}
      location={
        Object {
          "pathname": "/signin",
        }
      }
    >
      <Footer
        auth={
          Object {
            "allUsers": Object {
              "success": true,
              "users": Array [
                Object {
                  "avatar": null,
                  "bio": null,
                  "country": null,
                  "createdAt": "2018-03-13T05:04:37.443Z",
                  "email": "emasysnd@gmail.com",
                  "firstName": null,
                  "id": 1,
                  "lastName": null,
                  "moniker": "admin",
                  "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
                  "updatedAt": "2018-03-13T05:04:37.443Z",
                },
              ],
            },
            "authInfo": Object {
              "isLoggedIn": true,
              "userId": 1,
              "username": "admin",
            },
            "isLoggedIn": false,
            "message": null,
            "userInfo": Object {
              "data": Object {
                "avatar": null,
                "bio": null,
                "country": "Nigeria",
                "email": "emasysnd@gmail.com",
                "firstName": "emmy",
                "id": 1,
                "lastName": "endy",
                "moniker": "admin",
              },
              "success": true,
            },
            "userProfile": Object {
              "data": Object {
                "avatar": null,
                "bio": null,
                "country": null,
                "email": "emasysnd@gmail.com",
                "firstName": null,
                "id": 1,
                "lastName": null,
                "moniker": "admin",
              },
              "success": true,
            },
          }
        }
        flashMessage={[Function]}
        history={
          Object {
            "push": [Function],
          }
        }
        isAuthenticated={[Function]}
        location={
          Object {
            "pathname": "/signin",
          }
        }
      >
        <div>
          <footer
            className="container-fluid footer-main bg-dark"
          >
            <div
              className="container "
            >
              <div
                className="row "
              >
                <div
                  className="col-12 footer-social text-center mt-20"
                >
                  <a
                    href="https://web.facebook.com/emasys23"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-facebook-square text-white fa-2x"
                    />
                  </a>
                   
                  <a
                    href="https://twitter.com/emasys_nd"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-twitter-square text-white fa-2x"
                    />
                  </a>
                   
                  <a
                    href="https://github.com/emasys"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-github-square text-white fa-2x"
                    />
                  </a>
                  <p
                    className="text-center text-white"
                  >
                    © 
                    2018
                  </p>
                </div>
              </div>
            </div>
          </footer>
        </div>
      </Footer>
    </Authenticate>
  </Connect(Authenticate)>
</Provider>
`;

exports[`higher-order component withHoc should render without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Authenticate)>
    <Authenticate
      auth={
        Object {
          "allUsers": Object {
            "success": true,
            "users": Array [
              Object {
                "avatar": null,
                "bio": null,
                "country": null,
                "createdAt": "2018-03-13T05:04:37.443Z",
                "email": "emasysnd@gmail.com",
                "firstName": null,
                "id": 1,
                "lastName": null,
                "moniker": "admin",
                "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
                "updatedAt": "2018-03-13T05:04:37.443Z",
              },
            ],
          },
          "authInfo": Object {
            "isLoggedIn": true,
            "userId": 1,
            "username": "admin",
          },
          "isLoggedIn": true,
          "message": null,
          "userInfo": Object {
            "data": Object {
              "avatar": null,
              "bio": null,
              "country": "Nigeria",
              "email": "emasysnd@gmail.com",
              "firstName": "emmy",
              "id": 1,
              "lastName": "endy",
              "moniker": "admin",
            },
            "success": true,
          },
          "userProfile": Object {
            "data": Object {
              "avatar": null,
              "bio": null,
              "country": null,
              "email": "emasysnd@gmail.com",
              "firstName": null,
              "id": 1,
              "lastName": null,
              "moniker": "admin",
            },
            "success": true,
          },
        }
      }
      flashMessage={[Function]}
      isAuthenticated={[Function]}
    >
      <Footer
        auth={
          Object {
            "allUsers": Object {
              "success": true,
              "users": Array [
                Object {
                  "avatar": null,
                  "bio": null,
                  "country": null,
                  "createdAt": "2018-03-13T05:04:37.443Z",
                  "email": "emasysnd@gmail.com",
                  "firstName": null,
                  "id": 1,
                  "lastName": null,
                  "moniker": "admin",
                  "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
                  "updatedAt": "2018-03-13T05:04:37.443Z",
                },
              ],
            },
            "authInfo": Object {
              "isLoggedIn": true,
              "userId": 1,
              "username": "admin",
            },
            "isLoggedIn": true,
            "message": null,
            "userInfo": Object {
              "data": Object {
                "avatar": null,
                "bio": null,
                "country": "Nigeria",
                "email": "emasysnd@gmail.com",
                "firstName": "emmy",
                "id": 1,
                "lastName": "endy",
                "moniker": "admin",
              },
              "success": true,
            },
            "userProfile": Object {
              "data": Object {
                "avatar": null,
                "bio": null,
                "country": null,
                "email": "emasysnd@gmail.com",
                "firstName": null,
                "id": 1,
                "lastName": null,
                "moniker": "admin",
              },
              "success": true,
            },
          }
        }
        flashMessage={[Function]}
        isAuthenticated={[Function]}
      >
        <div>
          <footer
            className="container-fluid footer-main bg-dark"
          >
            <div
              className="container "
            >
              <div
                className="row "
              >
                <div
                  className="col-12 footer-social text-center mt-20"
                >
                  <a
                    href="https://web.facebook.com/emasys23"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-facebook-square text-white fa-2x"
                    />
                  </a>
                   
                  <a
                    href="https://twitter.com/emasys_nd"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-twitter-square text-white fa-2x"
                    />
                  </a>
                   
                  <a
                    href="https://github.com/emasys"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-github-square text-white fa-2x"
                    />
                  </a>
                  <p
                    className="text-center text-white"
                  >
                    © 
                    2018
                  </p>
                </div>
              </div>
            </div>
          </footer>
        </div>
      </Footer>
    </Authenticate>
  </Connect(Authenticate)>
</Provider>
`;
